From afbe66b5dba1d5626024972b9f7bce18d64f379e Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Sat, 13 Jul 2019 11:14:02 -0400
Subject: [PATCH 439/587] configs: arm64/bcm2711: Use CONFIG_BRCMSTB_THERMAL
 instead of CONFIG_BCM2835_THERMAL

The Raspberry Pi 4 uses the brcmstb thermal driver rather than brcm2835,
based on the device tree compatible string 'brcm,avs-tmon-bcm2838'. With
CONFIG_BRCMSTB_THERMAL enabled, reading temperature from
/sys/class/thermal/thermal_zone0/temp works as expected instead of
returning EINVAL.

Fixes: https://github.com/raspberrypi/linux/issues/3071
Signed-off-by: Allen Wild <allenwild93@gmail.com>
---
 arch/arm64/configs/bcm2711_defconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/configs/bcm2711_defconfig b/arch/arm64/configs/bcm2711_defconfig
index 3bcce6daa9ef..2265a77d6a53 100644
--- a/arch/arm64/configs/bcm2711_defconfig
+++ b/arch/arm64/configs/bcm2711_defconfig
@@ -658,7 +658,7 @@ CONFIG_SENSORS_ADS1015=m
 CONFIG_SENSORS_INA2XX=m
 CONFIG_SENSORS_TMP102=m
 CONFIG_THERMAL=y
-CONFIG_BCM2835_THERMAL=y
+CONFIG_BRCMSTB_THERMAL=y
 CONFIG_WATCHDOG=y
 CONFIG_GPIO_WATCHDOG=m
 CONFIG_BCM2835_WDT=y
-- 
2.17.1

